### PR TITLE
is_approx

### LINKED ISSUE #519: Generic inexact comparisons
From https://github.com/scipp/scipp/pull/517#issuecomment-529470900:

> I was thinking of a generic implementation that would just let you do similar(dataset1, dataset2, tolerance=1e-12, allow_transpose=True, allow_broadcast=True), or something like that.

### PR DESCRIPTION
**Pull request by OwenArnold**
Fixes #519

Basically use returning `transform` to perform an element by element `abs(x - y) < tol` type comparison and check all true. Light-weight function `is_approx` does work.

### CODE REVIEW

File: core/include/scipp/core/value_and_variance.h [hunk 1]
```diff
@@ -127,6 +127,22 @@ constexpr auto operator/(const T1 a, const ValueAndVariance<T2> b) noexcept {
...
+  return ValueAndVariance{a.value > b, a.variance};
+}
+template <class T1, class T2>
+constexpr auto operator>=(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value >= b, a.variance};
+}
+template <class T1, class T2>
+constexpr auto operator<=(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value <= b, a.variance};
+}
```
Comment (SimonHeybrock) [inline core/include/scipp/core/value_and_variance.h]: I think I would avoid adding these operators for `ValueAndVariance`. It is a bit ill-defined what they should do, I think, so making those operators accessible for all uses of `transform` is a bit dangerous. Instead, use a locally defined lambda/operator?

Comment (OwenArnold) [reply core/include/scipp/core/value_and_variance.h]: I agree. Particularly around the variance.

**Comment by SimonHeybrock** [comment]
@OwenArnold To avoid the issue with `transform` not supporting dropping of variances. Can we just use a variable inside the functor/lambda (captured by reference) to accumulate the `bool` result?

Something like:
```cpp
transform_in_place(a, b, [&match] (const auto &a, const auto &b) { match &= abs(...) < tolerance; });
```

**Comment by DanNixon** [comment]
Done in https://github.com/scipp/scipp/pull/673

**Pull request closed without merging**